In [ ]:
# Operators
# Arithmetic operators
first_num = 5
second_num = 3

print(first_num + second_num)
print(first_num - second_num)
print(first_num * second_num)
print(first_num / second_num)
print(first_num // second_num) # floor division or double division
print(first_num % second_num)
print(first_num ** second_num) # exponential

8
2
15
1.6666666666666667
1
2
125


In [ ]:
# Assignment operator
first_num = 25

# first_num = first_num + 10
print(first_num)
first_num += 10
print(first_num)
first_num -= 10
first_num *= 10
first_num /= 10
first_num //= 10
first_num %= 10
first_num **= 10
#print(first_num)

25
35


In [ ]:
 #comparison operators -> Boolean(True or False)
print(10 > 5)
print(10 >= 5)
print(10 < 5)
print(10 <= 5)
print(10 == 10)
print(10 == '10') # data type also will be checked will comparing

True
True
False
False
True
False


In [ ]:
# logical operators -> and, or, not

age = 19
voter_id = False

# print(age >= 18 and voter_id == True) # both condition need to be satisfied
# print(age >= 18 or voter_id == True) # any one condition need to be satisfied

# True -> False or vice versa => not
print(not(10 > 5))
print(not(10 < 5))

False
True


In [ ]:
# identity operators -> is, is not (address comparison)
# num = 10.5
# print(type(num) is int)
# print(type(num) is not int)

# in python memories are efficiently managed (upto 256 -> same address)
a = 256
b = 256
print("address of a", id(a))
print('address of b', id(b))
print (a is b)

a = 1
b = 2
print("address of a", id(a))
print('address of b', id(b))
print (a is b)

address of a 132284826165456
address of b 132284826165456
True
address of a 132284826157296
address of b 132284826157328
False


In [ ]:
# logical operators -> and, or, not

age = 19
voter_id = False

# print(age >= 18 and voter_id == True) # both condition need to be satisfied
# print(age >= 18 or voter_id == True) # any one condition need to be satisfied

# True -> False or vice versa => not
print(not(10 > 5))
print(not(10 < 5))

False
True


In [ ]:
# identity operators -> is, is not (address comparison)
# num = 10.5
# print(type(num) is int)
# print(type(num) is not int)

# in python memories are efficiently managed (upto 256 -> same address)
a = 256
b = 256
print("address of a", id(a))
print('address of b', id(b))
print (a is b)

a = 1
b = 2
print("address of a", id(a))
print('address of b', id(b))
print (a is b)

address of a 1899056425360
address of b 1899056425360
True
address of a 1899056228656
address of b 1899056228688
False


In [ ]:
# membership operator -> in, not in
news = 'India won the world cup 2023'
# print('India' in news)
# print('india' in news) # python is case sensitive

print('pakistan' not in news)

In [ ]:
# membership operator -> in, not in
news = 'India won the world cup 2023'
# print('India' in news)
# print('india' in news) # python is case sensitive

print('pakistan' not in news)

True


In [ ]:
# bitwise operators

# # bitwise and
# print(2 & 3)
# print(9 & 5)

# # bitwise or
# print(2 | 3)
# print(9 | 5)

# # bitwise XOR
# print(2 ^ 3)
# print(9 ^ 5)

# bitwise left shift, << (doubling)
# print(3 << 0) # num1 * 2^num2
# print(3 << 1)
# print(3 << 2)
# print(3 << 3)
# print(120 << 7)

# bitwise right shift (reducing by half)
# print(8 >> 0) # num1 // 2^num2
# print(8 >> 1)
# print(8 >> 2)
# print(8 >> 3)
# print(128 >> 5)
# print(120 >> 4)
# print(193 >> 4)

# bitwise not
print(~10)
print(~-10)

-11
9


In [ ]:
# remainder
# print(10 % 3)
# print(-10 % 3)

# remainder theorem
# rem = number - (divisor * quotient)

# floor division
# print(-10//3)


0
0


In [ ]:
# python handles upto 16 decimal points
import math
print(math.ceil(4.0000000000000001))
print(math.floor(4.9999999999999999))

4
5


In [ ]:
a,b,c = 10,20,50
#result = (a + b)//c
print("The result is {}".format(c))

In [ ]:
a = 10
b = 20
c = a**2
print(c)
d = b**2
print(d)
result = d - c
result -= 100
print(result)

In [ ]:
# while loop -> entry controlled loop
# 1 to 10

i = 1
while(i < 11):
    print(i)
    i += 1

1
2
3
4
5
6
7
8
9
10


In [ ]:
# terminal
while (True):
    command = input('C:/admin/user> ')
    if (command == 'add'): print('adding')
    elif command == 'sub': print('subtracting')
    elif command == 'mul': print('muiltiplying')
    elif command == 'quit':
        print('quitting...')
        break
    else: print('Invalid command')

C:/admin/user> add
adding
C:/admin/user> jjh
Invalid command
C:/admin/user> hj
Invalid command
C:/admin/user> quit
quitting...


In [ ]:
# function
def get_factorial(num):
    factorial = 1
    for i in range(1, num+1):
        factorial = factorial * i
#     print(factorial)
    return factorial

factorial = get_factorial(5)
print('factorial ->', factorial)

factorial -> 120


In [ ]:
def add(num1 = 0, num2 = 0):
    return num1 + num2

# print(add(20, 34))
# print(add(20, 34, 18)) # error

In [ ]:
# *args -> arguments
def add(*nums):
#     print(nums, type(nums))
    total = 0
    for num in nums:
        total += num
    return total

# add()
# add(10)
# add(10, 20)
add(10, 20, 10, 34, 68, 189)

331

In [ ]:
# **kwargs -> keyword arguments

def welcome(**details):
#     print(details, type(details))
    print(f"Welcome to Intellipaat {details.get('name', 'User')} and your id is {details.get('id_no', -1)}")

# welcome(id_no = 7, name = 'murali', place = 'india')
# welcome(id_no = 7)
welcome()

Welcome to Intellipaat User and your id is -1


In [ ]:
# first class functions or higher order functions

def circle(rad):
    return 3.14 * rad * rad

def square(side):
    return side * side

def rectangle(side1, side2):
    return side1 * side2

In [ ]:
def get_area(shape, *args): # hof
    return shape(*args)

print(get_area(circle, 5))
# print(get_area(square, 5))
# print(get_area(rectangle, 10, 5))

In [ ]:
# lambda
# def add(n1, n2):
#     return n1 + n2

# add = lambda n1, n2: n1 + n2
# print(add(10, 5))

# odd or even
is_even = lambda num: num % 2 == 0
print(is_even(10))
print(is_even(5))

True
False


In [ ]:
for i in range(5):
    for j in  range(i+1):
        print('*',end=" ")
    print('\n')


* 

* * 

* * * 

* * * * 

* * * * * 



In [ ]:
def num(n):
    num = 1
    for i in range(0, n):
        num = 1
        for j in range(0, i+1):
            print(num, end=" ")
            num = num + 1
        print("\r")
n = 5
num(n)

In [ ]:
def check(s1, s2):

    if(sorted(s1)== sorted(s2)):
        print("The strings are anagrams.")
    else:
        print("The strings aren't anagrams.")

s1 = input("Enter string1: ")
# input1: "listen"
s2 = input("Enter string2: ")
# input2: "silent"
check(s1, s2)

Enter string1: rty'
Enter string2: hjkgt778
The strings aren't anagrams.


#1 Find the given number is Armstrong number or not
1, 2, 3, 4, 5, 6, 7, 8, 9, 153, 370, 371, 407
For example, let's take the number 153:

1^3 + 5^3 + 3^3 = 1 + 125 + 27 = 153

Here's how the process works:

Take an integer.
Break it into its individual digits.
Raise each digit to the power of the total number of digits.
Sum up these powered digits.
If the sum equals the original number, then it is an Armstrong number.

In [ ]:
num = int(input("Enter a number: "))
# Input: 407
sum = 0
temp = num
while temp > 0:
   digit = temp % 10
   sum += digit ** 3
   temp //= 10
if num == sum:
   print(num,"is an Armstrong number")
else:
   print(num,"is not an Armstrong number")
# Output: 407 is an Armstrong number

Enter a number: 345
345 is not an Armstrong number
